cloudformation/028-cfn-splash-page.yaml

```
@@ -0,0 +1,441 @@
+            Scope: openid
+            SessionCookieName: AWSELBAuthSessionCookie
+            SessionTimeout: 28800
+            UserPoolArn: !ImportValue prod-auth-Cognito-UserPoolArn
+            UserPoolDomain: !ImportValue prod-auth-Cognito-UserPoolDomain
+            UserPoolClientId: !Ref cognitoAppClient
+        - TargetGroupArn: !Ref targetGroupExternalBrowser
+          Order: 2
+          Type: forward
+      Conditions:
+        - Field: host-header
+          HostHeaderConfig:
+            Values:
+              - !Ref ServiceHostName
+      ListenerArn: !ImportValue prod-general-infra-Lb-HttpsListenerArn
+      Priority: 271
+
+  listenerRuleExternalApi:
+    Type: AWS::ElasticLoadBalancingV2::ListenerRule
+    Properties:
+      Actions:
+        - TargetGroupArn: !Ref targetGroupExternalApi
+          Order: 1
+          Type: forward
+      Conditions:
+        - Field: host-header
+          HostHeaderConfig:
+            Values:
+              - !Ref ServiceHostName
+        - Field: http-header
+          HttpHeaderConfig:
+            HttpHeaderName: Authorization
+            Values:
+              - "Bearer *"
+      ListenerArn: !ImportValue prod-general-infra-Lb-HttpsListenerArn
+      Priority: 270
+
+  internalListenerRule:
+    Type: AWS::ElasticLoadBalancingV2::ListenerRule
+    Properties:
+      Actions:
+        - TargetGroupArn: !Ref internalTargetGroup
+          Order: 1
+          Type: forward
+      Conditions:
+        - Field: host-header
+          HostHeaderConfig:
+            Values:
+              - !Sub "${ServiceHostName}.internal"
+      ListenerArn: !ImportValue prod-general-infra-InternalLb-HttpListenerArn
+      Priority: 270
+
+  taskDef:
+    Type: AWS::ECS::TaskDefinition
+    Properties:
+      ContainerDefinitions:
+          - Name: "splash-page"
+          Essential: true
+          Image: !Ref ImageUri
+          LinuxParameters:
+            # https://docs.docker.com/engine/reference/run/#runtime-privilege-and-linux-capabilities
+            Capabilities:
+              ## Fargate does not allow adding Linux Capabilities. :-(
+              # Add:
+              #   - "CHOWN"
+              #   - "SETGID"
+              #   - "SETUID"
+              #   - "NET_BIND_SERVICE"
+              # Drop:
+              #   - "ALL"
+              Drop:
+                - SETPCAP
+                - MKNOD
+                - AUDIT_WRITE
+                - NET_RAW
+                - DAC_OVERRIDE
+                - FOWNER
+                - FSETID
+                - KILL
+                - SYS_CHROOT
+                - SETFCAP
+          LogConfiguration:
+            LogDriver: awslogs
+            Options:
+              awslogs-region: !Ref AWS::Region
+              awslogs-group: !Ref logGroup
+              awslogs-stream-prefix: openieui
+          MemoryReservation: 512
+          PortMappings:
+            - ContainerPort: 443
+              HostPort: 443
```

```suggestion
            - ContainerPort: 80
              HostPort: 80
```
The ALB will be doing SSL Termination, this should be running on port 80 assuming this is running on nginx or apache.